### PR TITLE
community: allow additional kwargs in MlflowEmbeddings for compatibility with Cohere API

### PR DESCRIPTION
**Pull request by elucherini**
- **Description:** add support for kwargs in`MlflowEmbeddings` `embed_document()` and `embed_query()` so that all the arguments required by Cohere API (and others?) can be passed down to the server.
  - **Issue:** #15234 
  - **Dependencies:** MLflow with MLflow Deployments (`pip install mlflow[genai]`)

**Tests**
Now this code [adapted from the docs](https://python.langchain.com/docs/integrations/providers/mlflow#embeddings-example) for the Cohere API works locally.

```python
"""
Setup
-----
export COHERE_API_KEY=...
mlflow deployments start-server --config-path examples/deployments/cohere/config.yaml

Run
---
python /path/to/this/file.py
"""
embeddings = MlflowCohereEmbeddings(target_uri="http://127.0.0.1:5000", endpoint="embeddings")
print(embeddings.embed_query("hello")[:3])
print(embeddings.embed_documents(["hello", "world"])[0][:3])
```

Output
```
[0.060455322, 0.028793335, -0.025848389]
[0.031707764, 0.021057129, -0.009361267]
```

### CODE REVIEW

**Review by hwchase17 — COMMENTED** [review]
i think prefereable to store as part of the class (so that we do not change the method signature)

**Comment by elucherini** [comment]
I just changed the code to this, let me know what you think:

```python

class MlflowEmbeddings(Embeddings, BaseModel):
    ...
    """Parameters to use for queries."""
    query_params: Dict[str, str] = {}
    """Parameters to use for documents."""
    documents_params: Dict[str, str] = {}

    ...

    def embed(self, texts: List[str], params: Dict[str, str]) -> List[List[float]]:
        # Mostly the same as the old embed_documents
        embeddings: List[List[float]] = []
        for txt in _chunk(texts, 20):
            # Add parameters to inputs
            resp = self._client.predict(endpoint=self.endpoint, inputs={"input": txt, **params})
            embeddings.extend(r["embedding"] for r in resp["data"])
        return embeddings

    def embed_documents(self, texts: List[str]) -> List[List[float]]:
        # Pass in documents_params
        return self.embed(texts, params=self.documents_params)

    def embed_query(self, text: str) -> List[float]:
        # Pass in query_params
        return self.embed([text], params=self.query_params)[0]
```

I also added a class with the specific parameters needed by Cohere:

```python
class MlflowCohereEmbeddings(MlflowEmbeddings):
    query_params: Dict[str, str] = {"input_type": "search_query"}
    document_params: Dict[str, str] = {"input_type": "search_document"}
```

This could expand to other frameworks, whether the additional parameters are required (like in this case) or optional.

File: libs/langchain/langchain/embeddings/mlflow.py [hunk 1]
```diff
@@ -1,3 +1,6 @@
-from langchain_community.embeddings.mlflow import MlflowEmbeddings
+from langchain_community.embeddings.mlflow import (
+    MlflowCohereEmbeddings,
+    MlflowEmbeddings,
+)
 
-__all__ = ["MlflowEmbeddings"]
+__all__ = ["MlflowEmbeddings", "MlflowCohereEmbeddings"]
```
Comment (hwchase17) [inline libs/langchain/langchain/embeddings/mlflow.py:6]: should not update code in `langchain` package

Comment (elucherini) [reply libs/langchain/langchain/embeddings/mlflow.py]: Fixed, only updating `community`

**Comment by elucherini** [comment]
@hwchase17 gentle reminder

**Review by baskaryan — APPROVED** [review]

**Pull request merged**